pkg/util/stringencoding/string_encoding.go

```diff
@@ -118,17 +125,21 @@ const uppercaseHex = `0123456789ABCDEF`
 // writeMultibyteRuneAsHex is equivalent to either
 // fmt.FPrintf(`\u%04X`) or fmt.FPrintf(`\U%08X`).
 // We can't quite just use strconv since we need uppercase hex.
-func writeMultibyteRuneAsHex(buf *bytes.Buffer, r rune, ln int) {
+func writeMultibyteRuneAsHex(buf Buffer, r rune, ln int) {
+	const mask = 0x0f
 	if ln == 2 {
-		buf.WriteString(`\u0000`)
+		buf.WriteString(`\u`)
```

Comment:
if you go this way, be sure to use buf.Grow to ensure the allocation takes place at most 1 time, like in the original code.